WordPress/pattern-directory
# Add UI for unlisting a pattern from the directory

Adds a way to change the status of a pattern to "unlisted" from within the block editor. An unlisted pattern can then be republished by clicking the standard Publish button. An unlisted pattern has a notice at the top of the editor explaining that it won't be included in the public pattern directory. This workflow is intended for moderators who need to review patterns that have been flagged by users as inappropriate.

This also enhances the Patterns list table so that the Unlisted table view has an accurate count of the number of unlisted posts, but unlisted posts are not included in the "All" view for the table.

Originally this was going to add a dropdown where you could choose between different post statuses. However, the Block Editor makes this awkward to accomplish since the built in statuses of draft, pending, and publish are all set from different parts of the UI. It also seemed to me that we didn't really need multiple custom statuses, and "Unlisted" could probably cover all of our use cases.

Fixes #107

### Screenshots

![Screen Capture on 2021-06-14 at 11-38-31](https://user-images.githubusercontent.com/916023/121942373-43be2f00-cd05-11eb-8637-6ee9fa642478.gif)

![Screen Shot 2021-06-14 at 11 41 37](https://user-images.githubusercontent.com/916023/121942601-84b64380-cd05-11eb-9e35-dadf9d59db11.png)

### How to test the changes in this Pull Request:

1. Create and publish a pattern
2. Go back to the Edit screen for the pattern and click the Unlist button
3. A notice should appear at the top of the block editor explaining that the pattern is now unlisted.
4. Click the Publish button to re-list the pattern.
5. Try unlisting patterns that are in the draft and pending statuses as well.
6. With at least one pattern marked as unlisted, go to the Patterns list table. Unlisted patterns should not be included in the "All" view, but should appear when specifically viewing Unlisted.
